modules/codegen/src/smithy4s/codegen/internals/IR.scala

```diff
@@ -426,6 +427,7 @@ private[internals] object TypedNode {
     def map[B](f: A => B): AltValueTN[B] = this match {
       case ProductAltTN(value) => ProductAltTN(f(value))
       case TypeAltTN(value)    => TypeAltTN(f(value))
+      case UnitAltTN           => UnitAltTN
```

Comment:
thought: this is my third implementation of the fix, I think it's the most principled as `AltValueTN` now more accurately mirrors the `UnionMember` ADT used on the schema side of things.